data-prepper-plugins/otel-metrics-source/src/main/java/com/amazon/dataprepper/plugins/source/otelmetrics/OTelMetricsSource.java

```diff
@@ -0,0 +1,151 @@
+/*
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  The OpenSearch Contributors require contributions made to
+ *  this file be licensed under the Apache-2.0 license or a
+ *  compatible open source license.
+ *
+ *  Modifications Copyright OpenSearch Contributors. See
+ *  GitHub history for details.
+ */
+
+package com.amazon.dataprepper.plugins.source.otelmetrics;
+
+import com.amazon.dataprepper.metrics.PluginMetrics;
+import com.amazon.dataprepper.model.annotations.DataPrepperPlugin;
+import com.amazon.dataprepper.model.buffer.Buffer;
+import com.amazon.dataprepper.model.configuration.PluginSetting;
+import com.amazon.dataprepper.model.record.Record;
+import com.amazon.dataprepper.model.source.Source;
+import com.amazon.dataprepper.plugins.certificate.CertificateProvider;
+import com.amazon.dataprepper.plugins.certificate.model.Certificate;
+import com.amazon.dataprepper.plugins.health.HealthGrpcService;
+import com.amazon.dataprepper.plugins.source.otelmetrics.certificate.CertificateProviderFactory;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.grpc.GrpcService;
+import com.linecorp.armeria.server.grpc.GrpcServiceBuilder;
+import io.grpc.protobuf.services.ProtoReflectionService;
+import io.opentelemetry.proto.collector.metrics.v1.ExportMetricsServiceRequest;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+
+@DataPrepperPlugin(name = "otel_metrics_source", pluginType = Source.class)
+public class OTelMetricsSource implements Source<Record<ExportMetricsServiceRequest>> {
```

Comment:
@dlvenable Thanks for the suggestion. We have addressed most of the other review comments and are now looking into migrating this PR towards the new Event model.

Reply:
Thank you @kmssap ! One idea I have on this is that we can keep the Source the way it is. And just update the `otel_metrics_raw_processor` to output `Record<Event>`. This would probably be the most straightforward approach to use Events.

Thus we'd have:

```
@DataPrepperPlugin(name = "otel_metrics_raw_processor", pluginType = Processor.class)
public class OTelMetricsStringProcessor extends AbstractProcessor<Record<ExportMetricsServiceRequest>, Record<Event>> {
```

It also would make sense to drop the "String" from the name and just name the processor: `OTelMetricsRawProcessor`

Reply:
@dlvenable Yes this sounds good. What do you think about subclassing JacksonEvent with a new class JacksonMetric and changing the signature of the Processor to output Record<Event> (or Record<Metric>, using a new Interface that extends Event). The input would remain Record<ExportMetricsServiceRequest> for the time being until #546 is merged to main. After this the mapping logic could be migrated to OtelProtoCodec so that the source would also emit Events exclusively.